packages/flutter_tools/templates/plugin/ios.tmpl/Classes/pluginClass.m.tmpl

```diff
@@ -1,25 +1,20 @@
 #import "{{pluginClass}}.h"
 
-@implementation {{pluginClass}} {
+@implementation {{pluginClass}}
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
+  FlutterMethodChannel* channel = [FlutterMethodChannel
+      methodChannelWithName:@"{{projectName}}"
+            binaryMessenger:[registrar messenger]];
+  {{pluginClass}}* instance = [{{pluginClass}} new];
```

Comment:
We should use `[[{{pluginClass}} alloc] init]` here, as you say. Or make this an instance method (`initWithRegistrar:`?), and have the generated registry `alloc` - `initWithRegistrar:`.

We used `[[{{pluginClass}} alloc] initWithController:]` before.